internal/elasticsearch/index/index.go

```diff
@@ -448,6 +883,8 @@ func resourceIndexRead(ctx context.Context, d *schema.ResourceData, meta interfa
 			return diag.FromErr(err)
 		}
 	}
+	// TODO: We ideally should set read settings to each field to detect changes
+	// But for now, setting it will cause unexpected diff for the existing clients which use `settings`
```

Comment:
I haven't thought this through entirely, but I expect we'd be able to handle this with `DiffSuppressFunc`. So suppress the diff of individual settings when they're unset, but set in the generic `settings` field, we could also do the inverse and suppress the diff of generic `settings` which only contain settings managed by the individual settings fields.

Reply:
Oh I see! Yeah it seems to work.  Let me try

Reply:
> // This is valid only for attributes of primitive types, because
   // DiffSuppressFunc itself is only compatible with primitive types.

I found that `DiffSuppressFunc` can only be used for primitive types, so it can't be used..!